.github/workflows/test.yml

```diff
@@ -23,10 +47,19 @@ jobs:
             exit 1
           fi
 
-      - name: Jest Unit Tests
-        run: npm run test
-        env:
-          CI: true
+      - name: Cache Jest cache
+        id: cache-jest-cache
+        uses: actions/cache@v1
+        with:
+          path: .jest-cache
+          key: ${{ runner.os }}-jest
+
+      - name: All Unit Tests
+        if: ${{ matrix.chunk < env.CHUNKS }}
+        run: npx jest --listTests --json | jq -cM '[_nwise(length / ${{ env.CHUNKS }} | ceil)]' | jq '.[${{ matrix.chunk }}] | .[] | @text' | xargs npm test
```

Comment:
tried locally and it happens if `matrix.chunk` is greater than last index i.e. 2 in this case (0, 1, 2)
But this line should not execute for matrix.chunk=3 since it's not less than total CHUNKS i.e. 3

xargs npm test will result into following command for chunk 0

```
jest "/mnt/c/Users/Zohaib/Projects/App/tests/unit/loginTest.js" "/mnt/c/Users/Zohaib/Projects/App/tests/unit/ReportUtilsTest.js" "/mnt/c/Users/Zohaib/Projects/App/tests/unit/fetchFreePlanVerifiedBankAccountTest.js" "/mnt/c/Users/Zohaib/Projects/App/tests/unit/MigrationTest.js" "/mnt/c/Users/Zohaib/Projects/App/tests/unit/awaitStagingDeploysTest.js" "/mnt/c/Users/Zohaib/Projects/App/tests/unit/createOrUpdateStagingDeployTest.js" "/mnt/c/Users/Zohaib/Projects/App/tests/unit/checkDeployBlockersTest.js" "/mnt/c/Users/Zohaib/Projects/App/tests/unit/NetworkTest.js" "/mnt/c/Users/Zohaib/Projects/App/tests/unit/RequestTest.js" "/mnt/c/Users/Zohaib/Projects/App/tests/unit/WalletTest.js"
```

Reply:
@parasharrajat  Are you running locally since this workflow never ran on github action since it needs approval

![image](https://user-images.githubusercontent.com/13637803/191788028-20fda1ae-3d77-4f83-856b-557e762a2fbd.png)

Reply:
@parasharrajat It's because you are creating 1 chunk ([[all files paths]]) and then accessing index 1 which is not there.

Based on code, `matrix.chunk` can never be equal to `env.CHUNKS` for this command since it's protected by `if` condition
```
- name: All Unit Tests
        if: ${{ matrix.chunk < env.CHUNKS }}
```

Reply:
And running pull request tests for last iteration
```
- name: Pull Request Tests
        # Pull request related tests will be run in separate runner in parallel.
        if: ${{ matrix.chunk == env.CHUNKS }}
        run: tests/unit/getPullRequestsMergedBetweenTest.sh
```

Reply:
Oh, I see. I will recheck